dist/index.js

```diff
@@ -17455,25 +17455,57 @@ module.exports = JSON.parse('[[[0,44],"disallowed_STD3_valid"],[[45,46],"valid"]
 var __webpack_exports__ = {};
 // This entry need to be wrapped in an IIFE because it need to be isolated against other modules in the chunk.
 (() => {
+const os = __nccwpck_require__(2037);
 const core = __nccwpck_require__(2186);
 const exec = __nccwpck_require__(1514);
 const tc = __nccwpck_require__(7784);
 const { Octokit } = __nccwpck_require__(5375);
 
-const baseDownloadURL = "https://github.com/digitalocean/doctl/releases/download"
-const fallbackVersion = "1.84.0"
+const baseDownloadURL = "https://github.com/digitalocean/doctl/releases/download";
+const fallbackVersion = "1.98.1";
 const octokit = new Octokit();
 
-async function downloadDoctl(version) {
-    if (process.platform === 'win32') {
-        const doctlDownload = await tc.downloadTool(`${baseDownloadURL}/v${version}/doctl-${version}-windows-amd64.zip`);
-        return tc.extractZip(doctlDownload);
+async function downloadDoctl(version, osType, osMachine) {
+    var platform = 'linux';
+    var arch = 'amd64';
+    var extension = 'tar.gz';
+
+    switch (osType) {
+        case 'Darwin': 
+            platform = 'darwin';
+            break;
+        case 'Windows_NT':
+            platform = 'windows';
+            extension = 'zip'
+            break;
+        case 'Linux':
+            platform = 'linux';
+            break;
+        default:
+            core.warning(`unknown platform: ${osType}; defaulting to ${platform}`);
+            break;
     }
-    if (process.platform === 'darwin') {
-        const doctlDownload = await tc.downloadTool(`${baseDownloadURL}/v${version}/doctl-${version}-darwin-amd64.tar.gz`);
-        return tc.extractTar(doctlDownload);
+
+    switch (osMachine) {
+        case 'arm64': 
+            arch = 'arm64';
+            break;
+        case 'x86_64':
+            arch = 'amd64';
+            break;
+        case 'i386':
+        case 'i686':
+            arch = '386';
+            break;
```

Comment:
Is this right? It's returning nothing for `i386`, and `386` for the `i686` case.

Reply:
Oops, this is the `dist/index.js` file, meant to comment this in `main.js`.

Reply:
Both `i386` and `i686` should return `386`.  It should fall through. See: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch 

And this code in the playground:

https://developer.mozilla.org/en-US/play?id=gXkXrCTbKRYlZOT1lwoSexa1HV69yvAH4zvx%2B4xxixjlDSxmoDOdLB21TMZ5P9JjgnJzKpqyinb1HlHM

Reply:
Despite my experience with JavaScript, I've never run into that `switch` syntax 😅 sounds good!